keras/layers/pooling/base_global_pooling1d.py

```diff
@@ -31,6 +31,21 @@ def __init__(self, data_format="channels_last", keepdims=False, **kwargs):
         self.data_format = conv_utils.normalize_data_format(data_format)
         self.keepdims = keepdims
 
+    def build(self, input_shape):
+        input_shape = tf.TensorShape(input_shape)
+        if self.data_format == "channels_last":
```

Comment:
In each case, you can simplify the code by having a single block that uses an abstract axis/axis_range value, instead of 2 copy/pasted blocks.

Reply:
So we can have a function that raises error when a tensor with dimension 0 at rduction axis is passed along with the axis/axis_range
`def check_axis_of_reduction (tensor, axis/axis_range):
    (check for 0)
    raise error if 0 is found

def build(self, input_shape):
    check_axis_of_reduction(input_shape, axes)
     ....
`

Reply:
Sure, sounds good. `validate_reduction_axis`.